grakn-graql/src/test/java/ai/grakn/graql/internal/gremlin/fragment/DirectIsaTest.java

```diff
@@ -0,0 +1,222 @@
+
+import static ai.grakn.graql.Graql.and;
+import static ai.grakn.graql.Graql.var;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class DirectIsaTest {
+
+    private static final Var x = var("x");
+    private static final Var y = var("y");
+    private static final Var z = var("z");
+
+    private static final Var superType = var("superType");
+    private static final Var subType = var("subType");
+
+    private static final String thingy = "thingy";
+    private static final String thingy0 = "thingy0";
+    private static final String thingy1 = "thingy1";
+    private static final String thingy2 = "thingy2";
+    private static final String thingy3 = "thingy3";
+    private static final String related = "related";
+
+    private EmbeddedGraknTx<?> tx;
+
+    @ClassRule
+    public static final SampleKBContext context = SampleKBContext.load(graph -> {
+
+        EntityType entityType0 = graph.putEntityType(thingy0);
+        EntityType entityType1 = graph.putEntityType(thingy1);
+        EntityType entityType2 = graph.putEntityType(thingy2);
+        EntityType entityType3 = graph.putEntityType(thingy3);
+
+        EntityType superType1 = graph.putEntityType(thingy)
+                .sub(entityType0)
+                .sub(entityType1);
+
+        Role role1 = graph.putRole("role1");
+        Role role2 = graph.putRole("role2");
+        Role role3 = graph.putRole("role3");
+        superType1.plays(role1).plays(role2).plays(role3);
+        entityType2.plays(role1).plays(role2).plays(role3);
+        entityType3.plays(role1).plays(role2).plays(role3);
+        RelationshipType relationshipType = graph.putRelationshipType(related)
+                .relates(role1).relates(role2).relates(role3);
+
+        Entity entity1 = entityType1.addEntity();
+        Entity entity2 = entityType2.addEntity();
+        Entity entity3 = entityType3.addEntity();
+        relationshipType.addRelationship()
+                .addRolePlayer(role1, entity1)
+                .addRolePlayer(role2, entity2)
+                .addRolePlayer(role3, entity3);
+    });
+
+    @Before
+    public void setUp() {
+        tx = context.tx();
+    }
+
+    @Test
+    public void testMatchSyntax() {
+        QueryBuilder queryBuilder = tx.graql();
+        Match matchQuery;
+        GetQuery getQuery;
+
+        matchQuery = queryBuilder.match(x.directIsa(thingy1));
+        assertEquals("match $x isa! thingy1;", matchQuery.toString());
+
+        matchQuery = queryBuilder.match(x.directIsa(y));
+        assertEquals("match $x isa! $y;", matchQuery.toString());
+
+        getQuery = queryBuilder.parse("match $x isa! thingy1; get;");
+        assertEquals(queryBuilder.match(x.directIsa(thingy1)), getQuery.match());
+
+        getQuery = queryBuilder.parse("match $x isa! $y; get;");
+        assertEquals(queryBuilder.match(x.directIsa(y)), getQuery.match());
+    }
+
+    @Test
+    public void testInsert() {
```

Comment:
You can split this test in two! I'd go for `testInsertSyntax` and `whenInsertDirectIsa_InsertsADirectInstanceOfAType`.